File: frontend/src/api/hookTemplate.ts
import request from '@/utils/request'

export interface HookTemplate {
  id: number
  name: string
  description: string
  module: string
  params: string
  timeout: number
  ignore_errors: boolean
  retries: number
  delay: number
  created_at: string
  updated_at: string
}

/** 获取所有钩子模板 */
export function getHookTemplatesApi(): Promise<HookTemplate[]> {
  return request.get('/hook-templates').then((res) => res.data.data)
}

/** 获取钩子模板详情 */
export function getHookTemplateApi(id: number): Promise<HookTemplate> {
  return request.get(`/hook-templates/${id}`).then((res) => res.data.data)
}

/** 创建钩子模板 */
export function createHookTemplateApi(data: Partial<HookTemplate>): Promise<HookTemplate> {
  return request.post('/hook-templates', data).then((res) => res.data.data)
}

/** 更新钩子模板 */
export function updateHookTemplateApi(id: number, data: Partial<HookTemplate>): Promise<void> {
  return request.put(`/hook-templates/${id}`, data).then((res) => res.data)
}

/** 删除钩子模板 */
export function deleteHookTemplateApi(id: number): Promise<void> {
  return request.delete(`/hook-templates/${id}`).then((res) => res.data)
}

File: backend/models/workflow/hook_template.go
package workflow

import (
	"time"

	"gorm.io/gorm"
)

// HookTemplate 可复用的钩子模板（全局管理，可被多个工作流的任务引用）
type HookTemplate struct {
	ID           uint           `json:"id" gorm:"primaryKey"`
	Name         string         `json:"name" gorm:"size:100;not null;uniqueIndex"`
	Description  string         `json:"description" gorm:"size:500"`
	Module       string         `json:"module" gorm:"size:50;not null;default:shell"`
	Params       string         `json:"params" gorm:"type:text"`
	Timeout      int            `json:"timeout" gorm:"default:0"`
	IgnoreErrors bool           `json:"ignore_errors" gorm:"default:false"`
	Retries      int            `json:"retries" gorm:"default:0"`
	Delay        int            `json:"delay" gorm:"default:0"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `json:"-" gorm:"index"`
}

func (HookTemplate) TableName() string { return "hook_templates" }
